python/readiness_check.py

```diff
@@ -0,0 +1,66 @@
+"""
+  Copyright (C) 2017-2018 Dremio Corporation
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+from functools import wraps
+
+import errno
+import os
+import signal
+import time
+
+import requests
+
+def timeout(seconds, error=os.strerror(errno.ETIMEDOUT)):
+    """
+    A timeout decorator for readiness check.
+    """
+    def decorator(func):
+        def timeout_handler(signum, frame):
+            raise TimeoutError(error)
+
+        def wrapper(*args, **kwargs):
+            signal.signal(signal.SIGALRM, timeout_handler)
+            signal.alarm(seconds)
+            try:
+                result = func(*args, **kwargs)
+            finally:
+                signal.alarm(0)
+            return result
+
+        return wraps(func)(wrapper)
+
+    return decorator
+
+# Set timeout to 3 minutes
+@timeout(180)
```

Comment:
Given that we have an early exit mechanism, we can probably set this to something like 5 minutes or more since this is the exit if something has gone unexpectedly wrong.

Reply:
Timeout has been changed to 5 minutes aka 300 seconds.